File: BackUpForABook/UIControls/TRClasses/TRCell.m
//
//  TRCell.m
//  ATreeRingMap
//
//  Created by Hui Wang on 5/23/14.
//  Copyright (c) 2014 AnnieLyticx. All rights reserved.
//

#import "TRCell.h"

// Represnts cell object class that holds the visual presentation information
@implementation TRCell

- (id)init
{
	self = [super init];
	if (self) {
        _displayStyle = CellValueShow_Flat;
        _cellShape = CellValueShape_Circle;
        _showValue = TRUE;
        _value = -1.0;
        _name = nil;
        _unit = nil;
        _text = nil;
        _displaySize = 8;
        _layer = nil;
        _metricColor = [UIColor colorWithRed:0.8 green:0.8 blue:0.8 alpha:1.0];
        _image = nil;
    }
    
    return self;
}
@end

File: BackUpForABook/UIControls/TRClasses_swift/TRCell.swift
//
//  TRCell.swift
//  TreeRingMap_Swift
//
//  Created by dingf on 17/3/17.
//  Copyright © 2017年 MH.dingf. All rights reserved.
//

import UIKit

class TRCell: NSObject {
    var layer: CALayer?
    var displayStyle: CellValueShow?
    var cellShape: CellValueShape?
    var withShadow: Bool?
    var withOutline: Bool?
    var displaySize: Int?
    var showValue: Bool?
    var value: Double?
    var unit: String?
    var name: String?
    var text: String?
    var metricColor: UIColor?
    var image: UIImage?
    
    override init() {
        super.init()
        displayStyle = .flat
        cellShape = .circle
        showValue = true
        value = -1.0
        name = nil
        unit = nil
        text = nil
        displaySize = 8
        layer = nil
        metricColor = UIColor(red: 0.8, green: 0.8, blue: 0.8, alpha: 1.0)
        image = nil
    }
}
